core/src/main/java/org/bitcoinj/crypto/DeterministicEntropy.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.crypto;
+
+import org.bitcoinj.base.internal.HexFormat;
+import org.bitcoinj.wallet.DeterministicSeed;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import static org.bitcoinj.crypto.ChildNumber.HARDENED_BIT;
+
+/**
+ * Utility class for working with BIP-85, generating wallets, etc. (<a href="https://github.com/bitcoin/bips/blob/master/bip-0085.mediawiki">https://github.com/bitcoin/bips/blob/master/bip-0085.mediawiki</a>).
+ * <p>
+ * See test class org.bitcoinj.crypto.DeterministicEntropyTest for examples on how to use this class.
+ */
+public class DeterministicEntropy {
+    private static final Logger logger = LoggerFactory.getLogger(DeterministicEntropy.class);
+
+    private static ChildNumber createHardenedChildNumber(int childNumber) {
+        return new ChildNumber(soften(childNumber), true);
+    }
+
+    private static int soften(int a) {
+        return a & ~HARDENED_BIT;
+    }
+
+    private static boolean hasHardenedBit(int a) {
+        return (a & HARDENED_BIT) != 0;
+    }
+
+    private static final ChildNumber BIP85_PATH_ROOT = createHardenedChildNumber(83696968);
+    private static final ChildNumber BIP39_APPLICATION_NUMBER = createHardenedChildNumber(39);
+
+
+    public enum Language {
+        English(0),
+        Japanese(1),
+        Korean(2),
+        Spanish(3),
+        ChineseSimplified(4),
+        ChineseTraditional(5),
+        French(6),
+        Italian(7),
+        Czech(8),
+        Portuguese(9);
+
+        private final int intValue;
+
+        Language(int intValue) {
+            this.intValue = intValue;
+        }
+
+        public ChildNumber childNumber() {
+            return createHardenedChildNumber(intValue);
+        }
+    }
+
+    public enum WordCount {
+        Twelve(12, 16),
+        Eighteen(18, 24),
+        TwentyFour(24, 32);
+
+        private final int intValue;
+        private final int bitCount;
+
+        WordCount(int intValue, int bits) {
+            this.intValue = intValue;
+            this.bitCount = bits;
+        }
+
+        public static WordCount forWordCount(int wordCount) {
```

Comment:
This isn't needed if you use the `.get(3)` method instead of `.children[3]` in the only caller.